flutter-studio/src/io/flutter/project/FlutterProjectModel.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.project;
+
+import com.android.tools.idea.observable.core.*;
+import com.android.tools.idea.wizard.model.WizardModel;
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.project.Project;
+import io.flutter.module.FlutterProjectType;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Note that a single instance of this class is shared among all the steps in the wizard.
+ *
+ * There is some inconsistency in which values are saved and which reset when switching from
+ * one project type to another. The AS New Module wizard has similar inconsistencies (as of beta 5).
+ *
+ * TODO: Add tests to simulate clicking Next/Previous buttons and choosing different project types.
+ * Others:
+ * Show progress during 'flutter create'.
```

Comment:
If we're not doing this currently, this would be an important one. The `flutter create` command can take a long time to run, depending on network conditions (it finishes by running a pub get).